shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -468,6 +469,21 @@
 bool FlutterPlatformViewsController::SubmitFrame(GrContext* gr_context,
                                                  std::shared_ptr<IOSContext> ios_context,
                                                  SkCanvas* background_canvas) {
+  if (merge_threads_) {
+    // Threads are about to be merged, we drop everything from this frame
+    // and possibly resubmit the same layer tree in the next frame.
+    // Before merging thread, we know the code is not running on the main thread. Assert that
+    FML_DCHECK(![[NSThread currentThread] isMainThread]);
+    picture_recorders_.clear();
+    composition_order_.clear();
+    return true;
+  }
+
+  // Any UIKit related code has to run on main thread.
+  // When on a non-main thread, we only allow the rest of the method to run if there is no
+  // Pending UIView operations.
+  FML_DCHECK([[NSThread currentThread] isMainThread] || !HasPendingViewOperations());
```

Comment:
I updated to use !HasPendingViewOperations() which I think is more accurate in this situation.